packages/cli/src/UserManagement/auth/jwt.ts

```diff
@@ -2,39 +2,27 @@
 /* eslint-disable import/no-cycle */
 
 import * as jwt from 'jsonwebtoken';
+import { Response } from 'express';
 import { JwtToken, PublicUserData } from '../Interfaces';
 import { User } from '../../databases/entities/User';
 import config = require('../../../config');
 
 export async function issueJWT(user: User): Promise<JwtToken> {
-	// Clean up null user data so it won't break during JWT generation
-	const optionalProperties = [
-		'email',
-		'firstName',
-		'lastName',
-		'password',
-		'personalizationAnswers',
-	];
-	optionalProperties.forEach((optionalProperty) => {
-		// @ts-ignore
-		if (!user[optionalProperty]) {
-			// @ts-ignore
-			// eslint-disable-next-line no-param-reassign
-			delete user[optionalProperty];
-		}
-	});
-	const { id, email, firstName, lastName, password, personalizationAnswers } = user;
+	const { id, email, password } = user;
 	const expiresIn = 14 * 86400000; // 14 days
 
 	const payload = {
 		id,
-		email,
-		firstName,
-		lastName,
-		password: password ? password.slice(Math.round(password.length / 2)) : undefined,
-		personalizationAnswers,
 	} as PublicUserData;
 
+	if (email) {
+		payload.email = email;
+	}
+
+	if (password) {
+		payload.password = password.slice(Math.round(password.length / 2));
```

Comment:
Is this storing half of the user's password in clear text in the JWT payload? 🤔 If so, I'd advise to store a salted hash instead.

Reply:
We are now adding md5 to the user password so it's not plain text.

Reply:
@krynble NIST has [deprecated](https://tools.ietf.org/id/draft-lvelvindron-tls-md5-sha1-deprecate-01.html) MD5 and SHA1. As this is dealing with password data, can I suggest using SHA-256?